File: portal/src/types/report.ts
import { ReportStatus } from "./index";

export interface Report {
  id: string;
  tenant_id: string;
  generated_by: string;
  title: string;
  period_start: string;
  period_end: string;
  terminal_ids: string[] | null;
  file_url: string | null;
  file_size_bytes: number | null;
  status: ReportStatus;
  created_at: string;
}

export interface ReportCreateRequest {
  title: string;
  period_start: string;
  period_end: string;
  terminal_ids?: string[];
  include_transcripts?: boolean;
  include_notes?: boolean;
  accent_color?: string;
}

export interface ReportDownloadResponse {
  download_url: string;
  expires_in: number;
}

File: backend/app/schemas/report.py
from datetime import datetime

from pydantic import BaseModel


class ReportCreate(BaseModel):
    title: str
    period_start: datetime
    period_end: datetime
    terminal_ids: list[str] | None = None
    include_transcripts: bool = False
    include_notes: bool = True
    accent_color: str | None = None


class ReportResponse(BaseModel):
    id: str
    tenant_id: str
    generated_by: str | None
    title: str
    period_start: datetime
    period_end: datetime
    terminal_ids: list | None
    file_url: str
    file_size_bytes: int | None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ReportListResponse(BaseModel):
    items: list[ReportResponse]
    total: int


class ReportDownloadResponse(BaseModel):
    download_url: str
    expires_in: int = 3600
